tests/src/test/scala/cats/tests/IndexedStoreTSuite.scala

```diff
@@ -0,0 +1,118 @@
+    // We only need a Functor to derive a Bifunctor for IndexedStoreT
+    implicit val F: Functor[NonEmptyListWrapper] = NonEmptyListWrapper.comonad
+    Bifunctor[IndexedStoreT[NonEmptyListWrapper, ?, Int, ?]]
+  }
+
+  {
+    implicit val F: Comonad[NonEmptyListWrapper] = NonEmptyListWrapper.comonad
+    implicit val FS: Functor[IndexedStoreT[NonEmptyListWrapper, String, Int, ?]] =
+      IndexedStoreT.catsDataFunctorRightForIndexedStoreT
+
+    checkAll("IndexedStoreT[NonEmptyListWrapper, String, Int, ?]",
+      FunctorTests[IndexedStoreT[NonEmptyListWrapper, String, Int, ?]].functor[Int, Int, Int])
+    checkAll("Functor[IndexedStoreT[NonEmptyListWrapper, String, Int, ?]]",
+      SerializableTests.serializable(Functor[IndexedStoreT[NonEmptyListWrapper, String, Int, ?]]))
+
+    Functor[IndexedStoreT[NonEmptyListWrapper, String, Int, ?]]
+  }
+
+  {
+    implicit val F: Comonad[NonEmptyListWrapper] = NonEmptyListWrapper.comonad
+    implicit val FS: Contravariant[IndexedStoreT[NonEmptyListWrapper, Int, ?, Int]] =
+      IndexedStoreT.catsDataContravariantForIndexedStoreT
+
+    checkAll("IndexedStoreT[NonEmptyListWrapper, Int, ?, Int]",
+      ContravariantTests[IndexedStoreT[NonEmptyListWrapper, Int, ?, Int]].contravariant[Int, Int, Int])
+    checkAll("Contravariant[IndexedStoreT[NonEmptyListWrapper, Int, ?, Int]]",
+      SerializableTests.serializable(Contravariant[IndexedStoreT[NonEmptyListWrapper, Int, ?, Int]]))
+
+    Contravariant[IndexedStoreT[NonEmptyListWrapper, Int, ?, Int]]
+  }
+
+  {
+    implicit val F: Comonad[NonEmptyListWrapper] = NonEmptyListWrapper.comonad
+    implicit val FS: Bifunctor[IndexedStoreT[NonEmptyListWrapper, ?, Int, ?]] = IndexedStoreT.catsDataBifunctorForIndexedStoreT
+
+    checkAll("IndexedStoreT[NonEmptyListWrapper, ?, Int, ?]",
+      BifunctorTests[IndexedStoreT[NonEmptyListWrapper, ?, Int, ?]].bifunctor[String, String, String, Int, Int, Int])
+    checkAll("Bifunctor[IndexedStoreT[NonEmptyListWrapper, ?, Int, ?]]",
+      SerializableTests.serializable(Bifunctor[IndexedStoreT[NonEmptyListWrapper, ?, Int, ?]]))
+
+    Bifunctor[IndexedStoreT[NonEmptyListWrapper, ?, Int, ?]]
+  }
+
+
+  checkAll("StoreId[Int, ?]", ComonadTests[StoreId[Int, ?]].comonad[Int, Int, Int])
+  checkAll("Comonad[StoreId[Long, ?]]", SerializableTests.serializable(Comonad[StoreId[Long, ?]]))
+}
+
+object IndexedStoreTSuite extends IndexedStoreTSuiteInstances {
+  implicit def storeEq[S: Eq: Arbitrary, A: Eq]: Eq[StoreId[S, A]] =
+    indexedStoreTEq[Eval, S, S, A]
+
+  val simpleStore: StoreId[(Int, Int), Int] =  StoreId(_._1, (1, 5))
+}
+
+sealed trait IndexedStoreTSuiteInstances {
+
+  implicit def indexedStoreTEq[F[_], I, A, B](implicit I: Arbitrary[I],
+                                              EqI: Eq[I],
+                                                FAB: Eq[F[A => B]]): Eq[IndexedStoreT[F, I, A, B]] =
+    Eq.by[IndexedStoreT[F, I, A, B], (F[A => B], I)](_.run)
+}
+
+/** This data type exists purely for testing.
```

Comment:
Does this class need to go in its own file?

Reply:
I think we have precedent here with `ListWrapper` being its own file, so I think it could go into its own file, yeah :)

Reply:
Cool 👍